NGitLab/Impl/LintClient.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using NGitLab.Models;
+
+namespace NGitLab.Impl
+{
+    internal class LintClient : ILintClient
+    {
+        private readonly API _api;
+
+        public LintClient(API api)
+        {
+            _api = api;
+        }
+
+        public Task<LintCI> LintGitLabCIYamlAsync(string projectId, string yamlContent, LintCIOptions options, CancellationToken cancellationToken = default)
+        {
+            var url = BuildLintCIUrl(projectId, options);
+            url = Utils.AddParameter(url, "content", yamlContent);
+
+            return _api.Post().ToAsync<LintCI>(url, cancellationToken);
+        }
+
+        public Task<LintCI> LintProjectGitLabCIYamlAsync(string projectId, LintCIOptions options, CancellationToken cancellationToken = default)
```

Comment:
The [doc](https://docs.gitlab.com/ee/api/lint.html#validate-a-projects-ci-configuration) calls this `Validate a project’s CI configuration`

```suggestion
        public Task<LintCI> ValidateProjectCIConfigurationlAsync(string projectId, LintCIOptions options, CancellationToken cancellationToken = default)
```